src/cli/cli_udp.cpp

```diff
@@ -136,33 +136,18 @@ otError UdpExample::ProcessSend(int argc, char *argv[])
 {
     otError       error;
     otMessageInfo messageInfo;
-    otMessage *   message              = NULL;
-    int           curArg               = 0;
-    bool          autoGenPayload       = false;
-    unsigned long autoGenPayloadLength = 0;
+    otMessage *   message       = NULL;
+    int           curArg        = 0;
+    uint16_t      payloadLength = 0;
+    PayloadType   payloadType   = kTypeText;
 
     memset(&messageInfo, 0, sizeof(messageInfo));
 
-    VerifyOrExit(argc == 1 || argc == 3 || argc == 4, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(argc >= 1 && argc <= 4, error = OT_ERROR_INVALID_ARGS);
```

Comment:
do we actually only support argc == 4 ?

Reply:
argc = [1,2,3,4]

Reply:
updated the doc to reflect all supported `argc` scenarios. thanks